src/oaklib/implementations/rustsim/rustsim_implementation.py

```diff
@@ -229,14 +220,29 @@ def pairwise_similarity(
             ancestor_information_content=max_ic,
         )
         sim.ancestor_information_content = max_ic
-        if subject_ancestors is None and isinstance(self, OboGraphInterface):
-            subject_ancestors = set(self.ancestors(subject, predicates=predicates))
-            subject_ancestors.add(subject)
-        if object_ancestors is None and isinstance(self, OboGraphInterface):
-            object_ancestors = set(self.ancestors(object, predicates=predicates))
-            object_ancestors.add(object)
-        if subject_ancestors is not None and object_ancestors is not None:
-            sim.jaccard_similarity = jaccard_similarity(subject_ancestors, object_ancestors)
+        if subject == object:
+            sim.jaccard_similarity = 1.0
+        else:
+            if not predicates:
```

Comment:
this is a pretty expensive block to do in the inner loop - can we not just do this in rust?

Reply:
or at least cache this

Reply:
Yes, probably should do this in Rust - I made a [PR](https://github.com/Knowledge-Graph-Hub/rustsim/pull/27) to do this